modules/apps/frontend-taglib/frontend-taglib-clay/src/main/java/com/liferay/frontend/taglib/clay/servlet/taglib/SelectTag.java

```diff
@@ -0,0 +1,190 @@
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.frontend.taglib.clay.servlet.taglib;
+
+import com.liferay.frontend.taglib.clay.internal.servlet.taglib.BaseContainerTag;
+import com.liferay.frontend.taglib.clay.servlet.taglib.util.SelectOption;
+import com.liferay.petra.string.StringPool;
+import com.liferay.portal.kernel.language.LanguageUtil;
+import com.liferay.portal.kernel.util.Validator;
+import com.liferay.taglib.util.TagResourceBundleUtil;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import javax.servlet.jsp.JspException;
+import javax.servlet.jsp.JspWriter;
+
+/**
+ * @author Kresimir Coko
+ */
+public class SelectTag extends BaseContainerTag {
+
+	@Override
+	public int doStartTag() throws JspException {
+		setAttributeNamespace(_ATTRIBUTE_NAMESPACE);
+
+		if (Validator.isNotNull(getLabel())) {
+			setDynamicAttribute(StringPool.BLANK, "aria-label", getLabel());
+		}
+
+		return super.doStartTag();
+	}
+
+	public String getLabel() {
+		return LanguageUtil.get(
+			TagResourceBundleUtil.getResourceBundle(pageContext), _label);
+	}
+
+	public String getName() {
+		return _name;
+	}
+
+	public List<SelectOption> getOptions() {
+		return _options;
+	}
+
+	public boolean isDisabled() {
+		return _disabled;
+	}
+
+	public boolean isMultiple() {
+		return _multiple;
+	}
+
+	public void setDisabled(boolean disabled) {
+		_disabled = disabled;
+	}
+
+	public void setLabel(String label) {
+		_label = label;
+	}
+
+	public void setMultiple(boolean multiple) {
+		_multiple = multiple;
+	}
+
+	public void setName(String name) {
+		_name = name;
+	}
+
+	public void setOptions(List<SelectOption> options) {
+		_options = options;
+	}
+
+	@Override
+	protected void cleanUp() {
+		super.cleanUp();
+
+		_disabled = false;
+		_label = null;
+		_multiple = false;
+		_name = null;
+		_options = null;
+	}
+
+	@Override
+	protected String getHydratedModuleName() {
+		return "frontend-taglib-clay/Select";
+	}
+
+	@Override
+	protected Map<String, Object> prepareProps(Map<String, Object> props) {
+		props.put("disabled", _disabled);
+
+		if (Validator.isNotNull(getLabel())) {
+			props.put("label", getLabel());
+		}
+
+		props.put("multiple", _multiple);
+		props.put("name", _name);
+		props.put("options", _options);
+
+		return super.prepareProps(props);
+	}
+
+	@Override
+	protected String processCssClasses(Set<String> cssClasses) {
+		cssClasses.add("form-control");
+
+		return super.processCssClasses(cssClasses);
+	}
+
+	@Override
+	protected int processStartTag() throws Exception {
+		super.processStartTag();
+
+		JspWriter jspWriter = pageContext.getOut();
+
+		if (Validator.isNotNull(_label)) {
+			jspWriter.write("<label>");
+			jspWriter.write(_label);
+			jspWriter.write("</label>");
+		}
+
+		jspWriter.write("<select ");
```

Comment:
I guess we need to do this just because of the `<label>` so we can't use the `baseContainer`...

Reply:
Yea, looking at the Soy tag, it renders the following:
```html
<div class="form-group">
    <label>Label</label>

    <select class="form-control">
        <option />
        <option />
        <option />
        <option />
    </select>
</div>
```

If I made it select then we wouldn't be able to have the `<label>`, and probably not even the `<div class="form-group">`. I started with `baseContainer` being set to `select` but quickly saw that it won't work.

Reply:
Okay, okay, that's good, then.